### DISCUSSION TITLE
Filtering Scalar list of JSON objects

### DISCUSSION THREAD

**Discussion by douglasrcjames** [prisma/prisma#20497] [Q&A] [answered]
In my Next.js 13 app, I am using react-hook-form for my forms, and the field arrays that are output from RHF are always an array of objects and not a flat object, so when I pass this array of objects over to Prisma, I am having a hard time searching for records in my database based on a value within the array of objects. In my example below, I have articles, that can have multiple categories. Each category is an object with a `name`. I want to be able to search for articles that match an name value in the list. Below is what I've tried but nothing is returned, which seems to be a short side of using postgres for this setup... Is there anyway I can achieve this function in my app? Like a work around? Or do I just need to hope and pray this is implemented? (https://github.com/prisma/prisma/issues/10879) (https://github.com/prisma/prisma/discussions/15711) In the meantime, I may be forced to convert an array of objects like this to a flat array of strings, which isn't ideal.

**Schema:**
```prisma
model Article {
    id              String    @id
    title                String
    categories      Json[]
}
```

**Request:**
```js
prisma.article.findMany({
  where: {
    categories: {
      path: ["name"],
      contains: "industry",
    }
  }
})
```

**Comment by ludralph** [comment, marked as answer]
Hi @douglasrcjames 👋 

It seems that you are trying to filter on the key value of an object inside an array. However, this feature is currently only supported by the MySQL database connector in Prisma. For PostgreSQL, you can only filter on the presence of entire JSON objects. See this example from the [documentation](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#json-object-arrays). 

Unfortunately, there doesn't seem to be a workaround for PostgreSQL at the moment. You might want to add a 👍 to the GitHub issue you mentioned and add your use case to help our engineers prioritize it.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
Did you had a chance to look at Raphael's response? Please let us know in case you still need any help. If not, please consider marking the discussion as answered/closed.

**Reply by douglasrcjames** [reply]
Thanks @ludralph ! I ended up just flattening and unflattening an array of objects to an array of strings and back again when dealing with that data manipulation.
